src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -0,0 +1,717 @@
+        ObjectEqual,
+        ObjectNotEqual,
+        ObjectVBEqual,
+        ObjectVBNotEqual,
+        ObjectLike,
+        ObjectGreaterEqual,
+        ObjectGreater,
+
+        EnumEqual,
+        EnumNotEqual,
+        EnumLess,
+        EnumLessEqual,
+        EnumGreaterEqual,
+        EnumGreater,
+
+        PointerEqual,
+        PointerNotEqual,
+        PointerLess,
+        PointerLessEqual,
+        PointerGreaterEqual,
+        PointerGreater,
+
+        DynamicEqual,
+        DynamicNotEqual,
+        DynamicLess,
+        DynamicLessEqual,
+        DynamicGreaterEqual,
+        DynamicGreater
+    }
+
+    /// <summary>
+    /// Represents a conversion operation.
+    /// </summary>
+    public interface IConversionExpression : IHasOperatorExpression
+    {
+        IExpression Operand { get; }
+        ConversionKind Conversion { get; }
+        /// <summary>
+        /// True if and only if the conversion is indicated explicity by a cast operation in the source code.
+        /// </summary>
+        bool IsExplicit { get; }
+    }
+
+    /// <summary>
+    /// Kinds of conversions.
+    /// </summary>
+    public enum ConversionKind
+    {
+        None,
+        /// <summary>
+        /// Conversion is defined by the underlying type system and throws an exception if it fails.
+        /// </summary>
+        Cast,
+        /// <summary>
+        /// Conversion is defined by the underlying type system and produces a null result if it fails.
+        /// </summary>
+        AsCast,
+        /// <summary>
+        /// Conversion has VB-specific semantics.
+        /// </summary>
+        Basic,
+        /// <summary>
+        /// Conversion has C#-specific semantics.
+        /// </summary>
+        CSharp,
+        /// <summary>
+        /// Conversion is implemented by a conversion operator method.
+        /// </summary>
+        Operator
+    }
+
+    public interface IConditionalChoiceExpression : IExpression
+    {
+        IExpression Condition { get; }
+        IExpression IfTrue { get; }
+        IExpression IfFalse { get; }
+    }
+
+    public interface INullCoalescingExpression : IExpression
+    {
+        IExpression Primary { get; }
+        IExpression Secondary { get; }
+    }
+
+    public interface IIsExpression : IExpression
+    {
+        IExpression Operand { get; }
+        ITypeSymbol IsType { get; }
+    }
+
+    public interface ITypeOperationExpression : IExpression
+    {
+        TypeOperationKind TypeOperationClass { get; }
+        ITypeSymbol TypeOperand { get; }
+    }
+
+    /// <summary>
+    /// Kinds of type operations.
+    /// </summary>
+    public enum TypeOperationKind
+    {
+        None,
+
+        SizeOf,
+        TypeOf
+    }
+
+    public interface ILambdaExpression : IExpression
+    {
+        IMethodSymbol Signature { get; }
+        IBlockStatement Body { get; }
+    }
+
+    public interface ILiteralExpression : IExpression
+    {
+        string Spelling { get; }
```

Comment:
Apparently I left quite a number of interfaces without doc comments. All need them.